### ISSUE TITLE
'num_iterations' keyward seem to be out of work

### ISSUE THREAD

**Issue by nerdylinius** [Microsoft/LightGBM#343]
I tried the following codes expected got a model with 120 trees, but got one with 100 trees. Is the 'num_iterations'  keyward not available here?

import lightgbm as lgb
from sklearn import datasets
import numpy as np

X, y = datasets.make_regression(n_samples=100000, n_features=100)

lgb_train = lgb.Dataset(X, y)

params = {
    'task': 'train',
    'boosting_type': 'gbdt',
    'objective': 'regression',
    'metric': {'l2', 'auc'},
    'num_leaves': 31,
    'learning_rate': 0.05,
    'feature_fraction': 0.9,
    'bagging_fraction': 0.8,
    'bagging_freq': 5,
    'verbose': 0,
    'num_iterations': 120
}

print('Start training...')
gbm = lgb.train(params,
                lgb_train)
print('Save model...')
gbm.save_model('model.txt')

**Comment by guolinke** [comment]
The training is controlled by python in Python package. 
So you should pass ```num_boosting_round``` to the ```lgb.train```. 
https://github.com/Microsoft/LightGBM/blob/master/docs/Python-API.md#trainparams-train_set-num_boost_round100-valid_setsnone-valid_namesnone-fobjnone-fevalnone-init_modelnone-feature_nameauto-categorical_featureauto-early_stopping_roundsnone-evals_resultnone-verbose_evaltrue-learning_ratesnone-callbacksnone

**Comment by nerdylinius** [comment]
thx, I thought it might be the same way as xgboost :+1:

**Comment by i3v** [comment]
Same thing. 
[There's](https://github.com/Microsoft/LightGBM/blob/c860efeac955d16a0d108c2e924252bffaa35053/docs/Parameters.md) even a `For python/R package, cannot use this parameters to control number of iterations` note. But somehow I've managed to not notice it and it took quite a long while to understand that I was using default `num_boosting_round` value. IMHO, it would be nice if lgbm would warn user if the parameter from the dictionary is ignored. Though, taking that there's only two of us... Probably this is not a real issue for most users.

**Comment by Laurae2** [comment]
@i3v I think there should be docs/examples in Python where `num_boost_round` is explicit. In R the parameter as a separate argument is required explicitly both in docs and in examples (`nrounds`), prefilled as `10`.

**Comment by i3v** [comment]
@Laurae2 
Indeed, there are some examples. 
E.g. [here](https://github.com/Microsoft/LightGBM/blob/55de8ea651aa4f3af80d78d0452593be4e13bada/docs/Python-intro.md):
```
num_round = 10
bst = lgb.train(param, train_data, num_round, valid_sets=[test_data])
```
`num_round` is used as a positional argument, even though it's actually optional. May be this is to highlight its importance.  So, formally, everything is fine, but still, IMHO, it's rather easy to make a mistake here. A rather hard-to-notice mistake. That's why, IMHO, a warning would be nice. But, again, IMHO, if there's just two users who found this misleading, it's probably not a real issue. (Unless number of reports in this topic would eventually increase.) Adding a warning here might be a big change.

Another [note](https://github.com/Microsoft/LightGBM/blob/c860efeac955d16a0d108c2e924252bffaa35053/docs/Parameters.md) about the same parameter, 
   > note: `num_tree` here equal with `num_iterations`. For multi-class, it actually learns `num_class * num_iterations` trees.

looks a bit strange as well. 

I've just submitted a tiny PR which makes the first note more visible and rephrases the second. Hope maintainers would consider my version more readable. And hope it would help others to not run into the same issue.

**Comment by Laurae2** [comment]
@i3v The following note is not strange:

> note: num_tree here equal with num_iterations. For multi-class, it actually learns num_class * num_iterations trees.

This is the expected behavior of gradient boosted trees: 1 model per class requires num_class * num_iterations trees. Each boosting iteration is actually doing one iteration per class model.

**Comment by i3v** [comment]
@Laurae2  yep, I do get the meaning of this note. Just like with the first one - it is correct, but it just _looks_ a bit strange:

1. The use of "here equal with". I don't say it's grammatically incorrect, but "is equal to" looks way more common. (And maybe it is better to just get rid of these words at all)
2. OK, `num_tree == num_iterations` but what's about `num_trees` and others?

**Comment by Laurae2** [comment]
@i3v 

1. There are a lot of grammatical issues to fix (some wordings are strange but are still understandable)

2. They are just aliases referring to a single parameter as outlined in the doc.

**Comment by i3v** [comment]
@Laurae2 

1. Yes, sure.
2. Yes, sure. I mean it just sounds strange. The `num_tree here equal with num_iterations` says "`num_tree` == `num_iterations`". But we already know that these two (and some others) are equal, cause they are aliases. But for some reason this phrase mentions just these two parameters, not all. And it looks a bit strange.

**Comment by i3v** [comment]
The PR is now merged. Hope it would help someone one day :).
Thanks, @Laurae2!
